internal/storage/storage_handle.go

```diff
@@ -39,22 +46,57 @@ type storageClient struct {
 	client *storage.Client
 }
 
-// NewStorageHandle returns the handle of Go storage client containing
-// customized http client. We can configure the http client using the
-// storageClientConfig parameter.
-func NewStorageHandle(ctx context.Context, clientConfig storageutil.StorageClientConfig) (sh StorageHandle, err error) {
+// Followed https://pkg.go.dev/cloud.google.com/go/storage#hdr-Experimental_gRPC_API to create the gRPC client.
+func createGRPCClientHandle(ctx context.Context, clientConfig *storageutil.StorageClientConfig) (sc *storage.Client, err error) {
+	if clientConfig.ClientProtocol != mountpkg.GRPC {
+		return nil, fmt.Errorf("wrong client-protocol requested: %s", clientConfig.ClientProtocol)
```

Comment:
nit: you can use message like  "client-protocol requested is not GRPC: %s". This way you will be able to differentiate from error in createHTTPClientHandle

Reply:
Make sense, I'll update the error message in the next PR.